packages/hardhat-core/test/internal/solidity/compiler/downloader.ts

```diff
@@ -316,6 +316,98 @@ describe("Compiler downloader", function () {
         });
       });
     });
+
+    describe("When downloading fails", function () {
+      describe("getCompilersList", function () {
+        it("retries on a failed download", async function () {
+          let downloadCount = 0;
+          const failingDownloader = (mockDownloader = new CompilerDownloader(
```

Comment:
This feels a bit weird. Can't you just assign to `mockDownloader` and use that?

Reply:
I have moved the more complicated download function up, and used it across the tests in this block.